gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -317,10 +324,61 @@ MacroExpander::expand_crate ()
   // expand module tree recursively
 
   // post-process
+  feature_gate ();
```

Comment:
I think this should rather be done in the `SessionManager::compile_crate` function. This is where the rest of our pipeline happens